weedcoco/repo/deposit.py

```diff
@@ -35,22 +36,26 @@ def get_all_existing_hash(repository_dir):
     }
 
 
-def setup_dataset_dir(repository_dir):
-    if not os.path.isdir(repository_dir):
-        os.mkdir(repository_dir)
-        dataset_dir = repository_dir / "dataset_1"
+def setup_dataset_dir(repository_dir, upload_id=None):
+    if upload_id is None:
+        if not os.path.isdir(repository_dir):
```

Comment:
There is a mix of `pathlib` and `os.path` here which is a bit surprising to the reader.

```suggestion
        if not repository_dir.is_dir():
```